### DISCUSSION TITLE
WindTerm 2.3.0 Prerelease

### DISCUSSION THREAD

**Discussion by kingToolbox** [kingToolbox/WindTerm#430] [General]
# Note:

**The MacOS and Linux versions are in preparation for release and will be released as the WindTerm_2.3.0_Prerelease_2 or WindTerm_2.3.0_Prerelease_3 version.**

# Usage:
### Windows
Please note that the **WindTerm_2.3.0_Prerelease_1_Windows_Portable_x86_64** version only supports Windows 10 19H1 (v10.0.18309) and later and Windows 11, other Windows versions should use the **WindTerm_2.3.0_Prerelease_1_Windows_Portable_x86_32** version.

### MacOS
Please open dmg file and drag WindTerm.app to the `Applications` folder.

### Linux
Please use the following command to run the WindTerm after downloading.
``` shell
tar -xzvf WindTerm_2.3.0_Prerelease_Linux_Portable.tar.gz && cd ./WindTerm_2.3.0 && chmod +x ./WindTerm && ./WindTerm
```

### Upgrade
For Windows and Linux, copy or move the **profiles** folder from the **old Windows/Linux version** to the **new Windows/Linux version** and all the sessions and configuration will be restored.

For MacOS, copy or move the **profiles** folder from the old **WindTerm.app/Contents/MacOS** to the new **WindTerm.app/Contents/MacOS** and all the sessions and configuration will be restored.

The **profiles** folder is cross-platform. 

# Known Issues:
- [ZMODEM] **Only support file names contains ASCII characters in Windows, otherwise the file name of the transfered file will be garbled after transmission, but it does not affect the correctness of the file content.** Linux and MacOS version does not have this problem.

# Changes in Version 2.3.0 Prerelease 1:

# Features:
- [APP] **Added Windows x86_64 version.** #213 
- [APP] **Use the `FreeType` font engine in the Widnows version to produce high-quality output of most vector and bitmap font formats.** #278 
- [APP] Update the third-party libraries and tools to the latest version: #237 
    - Qt **v6.2.2** (Used by the x86_64 Windows version)
    - Qt **v5.15.2** (Used by the x86_32 Windows version)
    - libssh **v0.9.6**
    - openssl **v1.1.3m**
    - pcre2 **v10.39**
    - spdlog **v1.9.2**
    - zstd **v1.5.1**
    - gsudo **v1.0.2**
    - vcxsrv **v1.20.14.0**
- [SHELL] **Supports running 64-bit windows shells, like 64-bit cmd, powershell and so on.**
- [SSH] **Use `VcXsrv` instead of `XMing` as the X server to improve compatibility, display and performance.** #157 
- [GUI] Supports showing or hiding `Toolbar` in the focus mode. #331 
- [GUI] Added `New Session` menu item in the `Session` pane. #424 
- [GUI] Added `Description` text edit in the `Session` dialog. #336 #423 

# Improvements:
- [GUI] **Better support for high DPI displays.** #80 #251
- [TERM] Improve the performance of parsing the control sequences.

# Bugs:
- [APP] **When a thread exits, the new thread may reuse the local thread variable of the old thread before it is deleted, which will cause a serious random crash.**
- [GUI] **The user interface is not displayed properly when the screen changed.** #230 #348 
- [GUI] When the `Explorer` pane is hidden and displayed again, the text in the list is displayed incomplete. #256 
- [SHELL] When using the shells started by `gsudo`, `gsudo` does not handle the key `ctrl+c` correctly. #48  

<hr /><em>This discussion was created from the release <a href='https://github.com/kingToolbox/WindTerm/releases/tag/'>WindTerm 2.3.0 Prerelease</a>.</em>

**Comment by funkpopo** [comment]
Well done! Issue #256 has been repaired totally! Using Win+Shift+←/→ for multi-screen handoff, the zoom delay has vanished.

**Comment by eculqui** [comment]
![imagen](https://user-images.githubusercontent.com/460391/152594176-75df27cd-2a5a-4b2f-a581-72bd9190e35d.png)
